src/pages/index.js

```diff
@@ -0,0 +1,306 @@
+import Api from "../utils/Api.js";
+
+const editProfileBtn = document.querySelector(".profile__edit-btn");
+
+const editProfileModal = document.querySelector("#edit-profile-modal");
+
+const editProfileCloseBtn = editProfileModal.querySelector(".modal__close-btn");
+
+const editProfileForm = editProfileModal.querySelector(".modal__form");
+const editProfileNameInput = editProfileModal.querySelector(
+  "#profile-name-input"
+);
+const editProfileDescriptionInput = editProfileModal.querySelector(
+  "#profile-description-input"
+);
+
+const newPostBtn = document.querySelector(".profile__add-btn");
+
+const newPostModal = document.querySelector("#new-post-modal");
+const newPostForm = newPostModal.querySelector(".modal__form");
+const newPostSubmitBtn = newPostModal.querySelector(".modal__submit-btn");
+const newPostCloseBtn = newPostModal.querySelector(".modal__close-btn");
+const editProfileSubmitBtn =
+  editProfileModal.querySelector(".modal__submit-btn");
+const imageCaptionInput = newPostModal.querySelector("#image-caption-input");
+const imageLinkInput = newPostModal.querySelector("#image-link-input");
+const profileNameEl = document.querySelector(".profile__name");
+const profileDescriptionEl = document.querySelector(".profile__description");
+const profileAvatarEl = document.querySelector(".profile__avatar");
+
+const previewModal = document.querySelector("#preview-modal");
+const previewCloseBtn = previewModal.querySelector(
+  ".modal__close-btn_type_preview"
+);
+
+const previewImageEl = previewModal.querySelector(".modal__image");
+const previewImageCaptionEl = previewModal.querySelector(".modal__caption");
+
+const cardTemplate = document.querySelector("#card-template");
+
+const cardsList = document.querySelector(".cards__list");
+// Avatar Modal
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitBtn = avatarModal.querySelector(".modal__submit-btn");
+const avatarCloseBtn = avatarModal.querySelector(".modal__close-btn");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+const avatarBtn = document.querySelector(".profile__avatar-btn");
+// Delete Modal
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".delete__form");
+const deleteCancelBtn = deleteModal.querySelector(".delete__cancel-btn");
+const deleteXBtn = deleteModal.querySelector(".modal__close-btn_type_delete");
+deleteCancelBtn.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+deleteXBtn.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+//overlay clicking
+editProfileModal.addEventListener("mousedown", handleOverlayClick);
+newPostModal.addEventListener("mousedown", handleOverlayClick);
+previewModal.addEventListener("mousedown", handleOverlayClick);
+avatarModal.addEventListener("mousedown", handleOverlayClick);
+const api = new Api({
```

Comment:
* The `confirmation` (**delete**) modal doesn't close when they click on the `overlay`